photongo/Readme.md

```diff
@@ -122,356 +154,484 @@ enum Role {
-pst, err := client.Post.Update(
-  post.New().ConnectAuthor(
-    user.Connect().Email("bob@prisma.io"),
+post, err := client.Post.Update.Where(
+  photon.Post.ID.Equals("cjsx2j8bw02920b25rl806l07"),
+).Data(
+  photon.Post.Author.Connect.Where(
+    photon.User.Email.Equals("bob@prisma.io"),
   ),
-  post.Where().ID("cjsx2j8bw02920b25rl806l07"),
-)
+).Exec(ctx)
 ```
 
-### UpdateMany
+#### UpdateMany
 
-#### Update three posts by their IDs
+##### Update three posts by their IDs
 
 ```go
-updated, err := client.Post.UpdateMany(
-  post.New().Published(true),
-  post.Where().IDIn(
+updated, err := client.Post.UpdateMany.Where(
+  photon.Post.ID.In([]string{
     "cjsyqxwqv000l0982p5qdq34p",
     "cjsyqxwqo000j0982da8cvw7o",
     "cjsyqxwr0000n0982cke8i5sc",
-  ),
-)
+  }),
+).Data(
+  photon.Post.Published.Set(true),
+).Exec(ctx)
 ```
 
-#### Update all posts where the title contains the given string
+##### Update all posts where the title contains the given string
 
 ```go
-updated, err = client.Post.UpdateMany(
-  post.New().Published(true),
-  post.Where().TitleContains("prisma"),
-)
+updated, err = client.Post.UpdateMany.Where(
+  photon.Post.Title.Contains("prisma"),
+).Data(
+  photon.Post.Published.Set(true),
+).Exec(ctx)
 ```
 
-### Delete
+#### Delete
 
-#### Delete a post by its ID
+##### Delete a post by its ID
 
 ```go
-pst, err := client.Post.Delete(post.Where().ID(10))
+post, err := client.Post.Delete.Where(
+  photon.Post.ID.Equals(10),
+).Exec(ctx)
 ```
 
-#### Delete a user by their email
+##### Delete a user by their email
 
 ```go
-usr, err := client.User.Delete(user.Where().Email("alice@prisma.io"))
+user, err := client.User.Delete.Where(
+  photon.User.Email.Equals("alice@prisma.io"),
+).Exec(ctx)
 ```
 
-### DeleteMany
+#### DeleteMany
 
-#### Delete all posts that were created before 2018:
+##### Delete all posts that were created before 2018:
 
 ```go
-deleted, err := client.Post.DeleteMany(
-  post.Where().CreatedAtGt(christmas),
-)
+deleted, err := client.Post.DeleteMany.Where(
+  photon.Post.CreatedAt.Gt(christmas),
+).Exec(ctx)
 ```
 
-### Upsert
+#### Upsert
 
-#### Create a user or update their role
+##### Create a user or update their role
 
 ```go
-usr, err := client.User.Upsert(
-  user.New().Email("alice@prisma.io"),
-  user.New().Role(user.Role.ADMIN),
-  user.Where().Email("alice@prisma.io"),
-)
+user, err := client.User.Upsert.Where(
+  photon.User.Email.Equals("alice@prisma.io"),
+).Data(
+  photon.User.Email.Set("alice@prisma.io"),
+  photon.User.Role.Set(user.Role.ADMIN),
+).Exec(ctx)
 ```
 
-### Select
+#### Select multiple things
 
-#### Select a user, their posts and their comments
-
-We're using generated fields here to maintain type-safety
+##### Select a user with 10 of their posts
 
 ```go
-var u struct {
-  user.ID
-  user.Email
-  Posts []struct {
-    post.Title
-    post.CreatedAt
-    Comments []struct {
-      comment.Comment
-      comment.Text
-    }
-  }
-}
-
-err := client.User.Select(&u,
```

Comment:
Do you have a proposal for Select?

Reply:
If you just want to select multiple things, you can query that by using the `.With()` methods.